### DISCUSSION TITLE
Does anyone have a video or know how to do a certificate for lets encrypt ?

### DISCUSSION THREAD

**Discussion by Nicc0110** [coder/code-server#3214] [Q&A]
i did it but now it says and idk how to use a cert with code-server.

Welcome to nginx!
If you see this page, the nginx web server is successfully installed and working. Further configuration is required.

For online documentation and support please refer to nginx.org.
Commercial support is available at nginx.com.

Thank you for using nginx.

**Comment by nooobcoder** [comment]
1) Use this to generate a self signed certificate in the present working directory.

```
openssl req \
       -newkey rsa:2048 -nodes -keyout certs.key \
       -x509 -days 365 -out certs.crt
```

2) If you are using a docker container, link the certificate files by binding them with the following locations of the container. See below,
`-v /home/AppConfig/CodeServer/certs/certs.crt:/home/coder/cert.crt `
`-v /home/AppConfig/CodeServer/certs/certs.key:/home/coder/cert.key`

3) Now, just after specifiying the image name (if you are using docker based deployment), use the two commands for the container to use the certificate files.

`codercom/code-server:latest --cert /home/coder/cert.crt --cert-key /home/coder/cert.key`
-------------------------------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This is the main command that specifies `code-server` to use the cert files.
`--cert /home/coder/cert.crt --cert-key /home/coder/cert.key`

**Reply by nooobcoder** [reply]
@Mcboy13 i assumed that you are using a docker based deployment, please let me know if anyhow I can help you further.

**Also, if you feel I have answered your question and your query has been resolved, then please mark the discussion as marked, this would help the moderators to mark discussions that do not required to be discussed farther.**

Thank You,
from **Ankur Paul** 🇮🇳

**Reply by Nicc0110** [reply]
I’m not using a docker I used the guide on the homepage of this and followed that. And it didn’t have anything to do with docker.

**Reply by nooobcoder** [reply]
@Mcboy13 are you talking about Coder or code-server
Let me tell you there are quite differences between the two.
Can you please confirm you are using code-server or Coder?

Also, which guide did you follow?
Also, code-server does not have a homepage, Coder does.
code-server has a github repo though.

Did you follow the guide to install code-server from the easy installation script from the documentation?
Did you follow this guide from the documentation?
![image](https://user-images.githubusercontent.com/50350828/116083923-d73a9280-a6ba-11eb-89a4-c00353aa7163.png)

**Reply by Nicc0110** [reply]
im using the one of the picture u sent!

**Comment by jsjoeio** [comment]
cc @bpmct - we should make a video about this!

**Comment by Nicc0110** [comment]
thanks i found out how to do it but my homepage says 

Welcome to nginx!
If you see this page, the nginx web server is successfully installed and working. Further configuration is required.

For online documentation and support please refer to nginx.org.
Commercial support is available at nginx.com.

Thank you for using nginx.


how do i fix this?

**Reply by Nicc0110** [reply]
@nooobcoder

**Reply by nooobcoder** [reply]
@Mcboy13 hey, sorry for late reply, I am replying you at 2:04 AM (local time) lol
See, first you should generate the cert files using the command,
```
openssl req \
       -newkey rsa:2048 -nodes -keyout certs.key \
       -x509 -days 365 -out certs.crt
```

Just type what comes along the screen and then get the path of the generated files, fyi the files are generated in your present directory.
Next, type `code-server --cert /home/coder/cert.crt --cert-key /home/coder/cert.key`
and replace **/home/coder/cert.crt** and **/home/coder/cert.key** with the absolute path to the generated cert files respectively. 
After all being done, reload the code-server service for the certificates to take effect.
Here is the command
`sudo systemctl restart code-server`

I hope this shall work now, let me know if any further problem occurs.
Thank You,
**Ankur Paul** 🇮🇳

**Reply by nooobcoder** [reply]
Also, when you use self signed certs, you would get HTTPS but still would not get the green lock icon as the certificates are not verified by a central body.
This is how it shall appear,
![image](https://user-images.githubusercontent.com/50350828/116147664-9f087380-a6fd-11eb-939c-90143acc6823.png)
But ya, you would have an URL that is accessible via **HTTPS** like this,
_https://192.168***********_

**Reply by Nicc0110** [reply]
hey thanks but my website says 
Welcome to nginx!
If you see this page, the nginx web server is successfully installed and working. Further configuration is required.

For online documentation and support please refer to nginx.org.
Commercial support is available at nginx.com.

Thank you for using nginx.

how do i fix this?

but i have to use a port so i can access how do i make it so i dont have to use port

**Reply by Nicc0110** [reply]
@nooobcoder

**Reply by nooobcoder** [reply]
@Mcboy13 show me the output of 
`sudo systemctl status code-server`

It should show similar output to indicate that it has got the cert files and is serving with HTTPS.
![image](https://user-images.githubusercontent.com/50350828/116180648-e8c18000-a736-11eb-987c-2ab0e7d9a1d8.png)

Also, I request you to check if you are hitting the correct port to access code-server.

Thank You

**Reply by code-asher** [reply]
Just a quick FYI: if you use `code-server --cert` (without a value) then code-server will generate self-signed certificates automatically! (So no need to run `openssl` with this method.) (It puts them in the data directory I believe.)

**Reply by nooobcoder** [reply]
@code-asher yes I have guided him regarding this, but he seems to have some Nginx error serving the default Nginx page and could not access code-server frontend.

**Reply by code-asher** [reply]
Also for setting up NGINX see https://github.com/cdr/code-server/blob/main/docs/guide.md#nginx. If you can I'd recommend using `--link` though, it does the proxying for you.
